session/manager.go

```
@@ -1,84 +1,167 @@
+	driverInstance, instanceExists = m.drivers[driverName]
+	if instanceExists {
+		return driverInstance, nil
 	}
 
-	if m.drivers[driverName] == nil {
-		return nil, errors.SessionDriverNotSupported.Args(driverName)
+	driverInstance = factory()
+	if driverInstance == nil {
+		return nil, errors.New(fmt.Sprintf("Session driver %s factory returned nil", driverName))
 	}
 
-	return m.drivers[driverName], nil
+	m.drivers[driverName] = driverInstance
+	m.startGcTimer(driverInstance) // Start GC for the new instance
+
+	return driverInstance, nil
 }
 
-func (m *Manager) Extend(driver string, handler func() sessioncontract.Driver) error {
-	if m.drivers[driver] != nil {
+// Extend registers a factory function for a given driver name.
+func (m *Manager) Extend(driver string, factory func() sessioncontract.Driver) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if _, exists := m.factories[driver]; exists {
 		return errors.SessionDriverAlreadyExists.Args(driver)
 	}
-	m.drivers[driver] = handler()
-	m.startGcTimer(m.drivers[driver])
+	m.factories[driver] = factory
 	return nil
 }
 
+// registerConfiguredDrivers ONLY reads from config.
+func (m *Manager) registerConfiguredDrivers() {
+	configuredDrivers := m.config.Get("session.drivers", map[string]any{})
+	driversMap, ok := configuredDrivers.(map[string]any)
+	if !ok {
+		return
+	}
+	for name, driverConfigAny := range driversMap {
+		driverConfig, ok := driverConfigAny.(map[string]any)
+		if !ok {
+
+			continue
+		}
+
+		viaFactoryAny, exists := driverConfig["via"]
+		if !exists {
+
+			continue
+		}
+
+		// Factory from config: func() (sessioncontract.Driver, error)
+		viaFactoryWithError, ok := viaFactoryAny.(func() (sessioncontract.Driver, error))
+		if !ok {
+
+			continue
+		}
+
+		// Wrapper for Extend signature: func() sessioncontract.Driver
+		factoryWrapper := func() sessioncontract.Driver {
+			driverInstance, err := viaFactoryWithError()
+			if err != nil {
+
+				return nil
```

Should return the error here.